src/main/java/core/util/HODateTime.java

```diff
@@ -23,50 +25,74 @@
     /**
      * the birthday of hattrick
      */
-    private static final HODateTime htStart = HODateTime.fromHT("1997-09-22 00:00:00");
+    public static final HODateTime htStart = HODateTime.fromHT("1997-09-22 00:00:00");
 
     /**
      * internal time representation
      */
-    public Instant instant;
+    final public Instant instant;
 
     /**
      * create an HODateTime instance (should it be private?)
+     *
      * @param in Instant
      */
-    public HODateTime(Instant in) {
+    public HODateTime(@NotNull Instant in) {
         this.instant = in;
     }
+    public HODateTime(@NotNull HODateTime in){this.instant=in.instant;}
 
     /**
      * Create instance from HT (chpp) string
+     *
      * @param htString HT string
      * @return HODateTime
      */
     public static HODateTime fromHT(String htString) {
-        LocalDateTime htTime = LocalDateTime.parse(htString, cl_Formatter);
-        return new HODateTime(htTime.atZone(DEFAULT_TIMEZONE).toInstant());
+        if ( htString != null && !htString.isEmpty()) {
+            LocalDateTime htTime = LocalDateTime.parse(htString, cl_Formatter);
+            return new HODateTime(htTime.atZone(DEFAULT_TIMEZONE).toInstant());
+        }
+        return null;
     }
 
     /**
      * Create an instance from database timestamp
+     *
      * @param timestamp database timestamp
      * @return HODateTime
      */
     public static HODateTime fromDbTimestamp(Timestamp timestamp) {
```

Comment:
One thing that I am concerned about is that there is no handling of timezones for some of these: all datetimes coming from CHPP seem to be CET/CEST-based (i.e. Europe/Stockholm).  For example this could be problematic for `toLocaleDate`?

Reply:
You're right: CHPP is CET/CEST (fromHT is intended to read this - as far i understand that correct, the resulting Instant is Zulu)
to and fromDbTimestamp should convert this to hsqldb timestamp ( i wondered that this works even if the table columns were defined as varchars - but i'm not very happy with that. dbupdate could change those columns to TIMESTAMP)

if there are timestamps in db which were stored in a different time zone, we could get into trouble. This i would like to handle in a dbupdate procedure.

strategy should be: Store all timestamps with timezone zulu.